frontend/src/components/widgets/DataFrame/columns/ChartColumn.ts

```diff
@@ -0,0 +1,201 @@
+ */
+function BaseChartColumn(
+  kind: string,
+  props: BaseColumnProps,
+  chart_type: "line" | "bar"
+): BaseColumn {
+  const parameters = mergeColumnParameters(
+    // Default parameters:
+    {
+      y_min: 0,
+      y_max: 1,
+    },
+    // User parameters:
+    props.columnTypeOptions
+  ) as ChartColumnParams
+
+  const cellTemplate = {
+    kind: GridCellKind.Custom,
+    allowOverlay: false,
+    copyData: "",
+    contentAlign: props.contentAlignment,
+    data: {
+      kind: "sparkline-cell",
+      values: [],
+      displayValues: [],
+      graphKind: chart_type,
+      yAxis: [parameters.y_min, parameters.y_max],
+    },
+  } as SparklineCellType
+
+  return {
+    ...props,
+    kind,
+    sortMode: "default",
+    isEditable: false, // Chart column is always read-only
+    getCell(data?: any): GridCell {
+      if (
+        isNullOrUndefined(parameters.y_min) ||
+        isNullOrUndefined(parameters.y_max) ||
+        Number.isNaN(parameters.y_min) ||
+        Number.isNaN(parameters.y_max) ||
+        parameters.y_min >= parameters.y_max
+      ) {
+        return getErrorCell(
+          "Invalid min/max y-axis configuration",
+          `The y_min (${parameters.y_min}) and y_max (${parameters.y_max}) configuration options must be valid numbers.`
+        )
+      }
+
+      if (isNullOrUndefined(data)) {
+        // TODO(lukasmasuch): Use a missing cell?
+        return getEmptyCell()
+      }
+
+      const chartData = toSafeArray(data)
+
+      const convertedChartData: number[] = []
+      let normalizedChartData: number[] = []
+      if (chartData.length === 0) {
+        return getEmptyCell()
+      }
+
+      // Initialize with smallest and biggest number
+      let maxValue = Number.MIN_SAFE_INTEGER
+      let minValue = Number.MAX_SAFE_INTEGER
+
+      // Try to convert all values to numbers and find min/max
+      for (let i = 0; i < chartData.length; i++) {
+        const convertedValue = toSafeNumber(chartData[i])
+        if (
+          Number.isNaN(convertedValue) ||
+          isNullOrUndefined(convertedValue)
+        ) {
+          return getErrorCell(
+            toSafeString(chartData),
+            `The value cannot be interpreted as a numeric array. ${toSafeString(
+              convertedValue
+            )} is not a number.`
+          )
+        }
+
+        if (convertedValue > maxValue) {
+          maxValue = convertedValue
+        }
+
+        if (convertedValue < minValue) {
+          minValue = convertedValue
+        }
+
+        convertedChartData.push(convertedValue)
+      }
+
+      if (
+        convertedChartData.length > 0 &&
+        (maxValue > parameters.y_max || minValue < parameters.y_min)
+      ) {
+        // Normalize values between the configured range
+        normalizedChartData = convertedChartData.map(
+          v =>
+            ((parameters.y_max || 1) - (parameters.y_min || 0)) *
+              ((v - minValue) / (maxValue - minValue)) +
```

Comment:
Do we have to worry about a divide by 0 error here?

Reply:
Added a fix